### PR TITLE
#184760040 Added login via Google and login using email and password

### PR DESCRIPTION
**Pull request by Marthatwesi**
### What does this PR do?
Login via Google and Login with email and password

### Any background context you want to provide?

N/A

### Questions:

# Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code is formatted correctly
- [x] My changes doesn't have any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/googleButton.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// components in this folder
+import React from 'react'
+import { Link } from 'react-router-dom';
+import "../styles/login.scss";
+import googleIcon from "../images/googleIcon.svg"
+export default function GoogleButton() {
+  return (
+    <div>
+      <Link to= "http://localhost:4000/api/v1/users/auth/google">
```
Comment (Eli250) [inline src/components/googleButton.js]: Remove the hard-coded URL. Protect your API with env

**Review by Ntarecedrick — CHANGES_REQUESTED** [review]
the login with google they are giving an error 400: redirect url mis_match you can maybe check if the deployed backend is working

File: src/components/loginInputs.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  const { loading,error,user } = useSelector(getLoginUser);
+  const [userObj, setUserObj] = useState({
+    email: "",
+    password: "",
+  });
+  const submitHandler = (data) => {
+    dispatch(loginThunk(data));
+  };
+  if (loading === false && user !== null) {
+    localStorage.setItem("t", user);
```
Comment (Eli250) [inline src/components/loginInputs.js]: What is `t` here?

File: src/pages/homepage.js [hunk 3]
```diff
@@ -1,6 +1,13 @@
-import React from "react";
+import React, { useEffect } from "react";
 
 const Homepage = () => {
+  useEffect(()=>{
+ if(window.location.href.includes("token")){
+const token = window.location.href.split("token=")[1]
+localStorage.setItem("t",token)
```
Comment (Eli250) [inline src/pages/homepage.js]: Use clear naming for the variables you are setting in the localStarage object.

**Review by Eli250 — CHANGES_REQUESTED** [review]

**Review by teerenzo — CHANGES_REQUESTED** [review]
you can add eye icon to see entered password

**Review by Eli250 — APPROVED** [review]

**Pull request merged**